crates/nu-command/src/env/export_env.rs

```diff
@@ -31,10 +30,15 @@ impl Command for ExportEnv {
         call: &Call,
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        let capture_block: Closure = call.req(engine_state, caller_stack, 0)?;
-        let block = engine_state.get_block(capture_block.block_id);
+        let block_id = call
+            .positional_nth(0)
+            .expect("checked through parser")
+            .as_block()
+            .expect("internal error: missing block");
+
+        let block = engine_state.get_block(block_id);
         let mut callee_stack = caller_stack
-            .captures_to_stack(capture_block.captures)
+            .gather_captures(engine_state, &block.captures)
```

Comment:
In this context is it semantically closer to a closure or do we do extra changes to allow the rest of block behavior like mutation?

Reply:
Probably closer to a closure, I like to think of it as a `def --env` call. So, local mutation is allowed, but it shouldn't be able to access any outside variables since it must be in a module.